src/main/java/edu/usf/sas/pal/muser/manager/MuserMusicDataAnalysisManager.java

```diff
@@ -45,15 +48,27 @@ public MuserMusicDataAnalysisManager() throws FirebaseFileNotInitializedExceptio
     public void processData() {
         // create csv file and add the header
         csvFileWriter.createHeader(MusicAnalysisModel.CSV_HEADER);
-        if (programOptions.getUserId() == null) {
+        // it will enter if loop if only date range is provided 
+        if (programOptions.getStartDate() != null && programOptions.getEndDate() != null && programOptions.getUserId() == null) {
+            filterUsersDataByDataRange(getStartDateInMillis(), getEndDateInMillis());
+        }
+        // it will enter this else if loop if a specific user date range is required
```

Comment:
Normal Java convention would be to have the code on one line and put the comment within the IF block if it's needed, like:

```
} else if (...
    // it will...
```